### PR TITLE
Implements activity streak API

### LINKED ISSUE #9078: [Referral]: Add API for the activity streak
- activity streak 

- [ ] Vesting
  - [ ] reward_distribution_bonus_multiplier

- [ ] Activity streak info

  - [ ] Inactivity_streak (ie. N epochs)
  - [ ] Activity_streak (ie. N epochs)
  - [ ] Reward_distribution_activity_multiplier
  - [ ] Reward_vesting_activity_multiplier

### PR DESCRIPTION
**Pull request by jeremyletang**
close #9078

### CODE REVIEW

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 1]
```diff
@@ -3328,6 +3335,20 @@ message ListEntitiesResponse {
...
+message GetPartyActivityStreakRequest {
+  // The party ID.
+  string party_id = 1 [(google.api.field_behavior) = REQUIRED];
+  // An optional epoch, returns last if not specified.
+  optional uint64 epoch = 2;
+}
+
+// Response to get a party activity streaks accross epochs
+message GetPartyActivityStreakResponse {
+  // The activity streak of a party at a given epoch
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:3348]:
```suggestion
  // A party's activity streak in a given epoch
```

File: protos/sources/vega/events/v1/events.proto [hunk 2]
```diff
@@ -29,6 +29,10 @@ message PartyActivityStreak {
   string reward_vesting_activity_multiplier = 6;
   // The epoch when these informations were accurate.
   uint64 epoch = 7;
+  // Traded volume for the party at the end of that given epoch
```
Comment (candida-d) [inline protos/sources/vega/events/v1/events.proto:32]:
```suggestion
  // Party's traded volume at the end of the given epoch
```

File: protos/sources/vega/events/v1/events.proto [hunk 3]
```diff
@@ -29,6 +29,10 @@ message PartyActivityStreak {
...
+  string traded_volume = 8;
+  // Open interest volume of the party at the end of the epoch
```
Comment (candida-d) [inline protos/sources/vega/events/v1/events.proto:34]:
```suggestion
  // Party's open interest volume at the end of the given epoch
```

File: protos/sources/vega/events/v1/events.proto [hunk 4]
```diff
@@ -29,6 +29,10 @@ message PartyActivityStreak {
   string reward_vesting_activity_multiplier = 6;
   // The epoch when these informations were accurate.
```
Comment (candida-d) [inline protos/sources/vega/events/v1/events.proto:30]:
```suggestion
  // The epoch this information is relevant to.
```

File: datanode/gateway/graphql/schema.graphql [hunk 5]
```diff
@@ -2358,6 +2358,34 @@ type Party {
     "Pagination information"
     pagination: Pagination
   ): TransferConnection
+
+  "The activity streak"
+  activityStreak(
+    "An option epoch"
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:2364]:
```suggestion
    "An optional epoch"
```

File: datanode/gateway/graphql/schema.graphql [hunk 6]
```diff
@@ -2358,6 +2358,34 @@ type Party {
...
+
+"""
+The activity streak for a party.
+"""
+type PartyActivityStreak {
+  "The number of epochs the party has been active in a row"
+  activeFor: Int!
+  "The number of epochs the party has been inactive in a row"
+  inactiveFor: Int!
+  "Is the party considered as active, and then eligible for rewards multipliers"
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:2377]:
```suggestion
  "If the party is considered as active, and thus eligible for rewards multipliers"
```

File: datanode/gateway/graphql/schema.graphql [hunk 7]
```diff
@@ -2358,6 +2358,34 @@ type Party {
...
+  isActive: Boolean!
+  "The rewards distribution multiplier for the party"
+  rewardDistributionMultiplier: String!
+  "The rewards vesting multiplier for the party"
+  rewardVestingMultiplier: String!
+  "The epoch at which this informations were valid"
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:2383]:
```suggestion
  "The epoch for which this information is relevant"
```

File: datanode/gateway/graphql/schema.graphql [hunk 8]
```diff
@@ -2358,6 +2358,34 @@ type Party {
...
+  epoch: Int!
+  "The traded volume for that party on that given epoch"
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:2385]:
```suggestion
  "The traded volume for that party in the given epoch"
```

File: datanode/gateway/graphql/schema.graphql [hunk 9]
```diff
@@ -2358,6 +2358,34 @@ type Party {
...
+  tradedVolume: String!
+  "The open volume for that party on that given epoch"
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:2387]:
```suggestion
  "The open volume for the party in the given epoch"
```

**Review by candida-d — DISMISSED** [review]

File: datanode/gateway/graphql/party_activity_streak.go
Comment (guoguojin) [inline datanode/gateway/graphql/party_activity_streak.go:1]: Not a deal breaker, but generally for resolvers, we name the files xx_resolver.go

**Review by guoguojin — DISMISSED** [review]

File: datanode/gateway/graphql/party_activity_streak.go
Comment (jeremyletang) [reply datanode/gateway/graphql/party_activity_streak.go:1]: alright, will fix.

**Review by guoguojin — APPROVED** [review]

**Pull request merged**